service/history/chasm_engine.go

```diff
@@ -141,14 +141,14 @@ func (e *ChasmEngine) NewExecution(
 		newExecutionParams,
 	)
 	if err != nil {
-		return chasm.ExecutionKey{}, nil, err
+		return chasm.ExecutionKey{}, nil, false, err
 	}
 	if !hasCurrentRun {
 		serializedRef, err := newExecutionParams.executionRef.Serialize(e.registry)
 		if err != nil {
-			return chasm.ExecutionKey{}, nil, err
+			return chasm.ExecutionKey{}, nil, true, err
```

Comment:
nit: false

Reply:
hmmm yeah good point, can you help add a comment on that. Thanks!

with that said the flag won't be accurate anyway, like on L144 above the execution may actually be created despite an error is returned.